consumers/client-event-consumer/src/main/java/com/flipkart/aesop/event/AbstractEventFactory.java

```diff
@@ -41,48 +39,32 @@
  */
 public abstract class AbstractEventFactory<T extends AbstractEvent> implements EventFactory
 {
-	public static String PRIMARY_KEY_FIELD_NAME = "pk";
-	public static String META_FIELD_TYPE_NAME = "dbFieldType";
-
-	/**
-	 * Generates primary key set using the schema.
-	 * @param schema
-	 * @return Primary key set
-	 * @throws DatabusException
-	 */
-	private Set<String> getPrimaryKeysSetFromSchema(Schema schema) throws DatabusException
-	{
-		Set<String> primaryKeySet = new HashSet<String>();
-		String primaryKeyFieldName = SchemaHelper.getMetaField(schema, PRIMARY_KEY_FIELD_NAME);
-		if (primaryKeyFieldName == null)
-		{
-			throw new DatabusException("No primary key specified in the schema");
-		}
-		for (String primaryKey : primaryKeyFieldName.split(DbusConstants.COMPOUND_KEY_SEPARATOR))
-		{
-			primaryKeySet.add(primaryKey.trim());
-		}
-		assert (primaryKeySet.size() >= 1);
-		return primaryKeySet;
-	}
-
 	public AbstractEvent createEvent(DbusEvent dbusEvent, DbusEventDecoder eventDecoder) throws DatabusException
 	{
 		GenericRecord genericRecord = eventDecoder.getGenericRecord(dbusEvent, null);
 		VersionedSchema writerSchema = eventDecoder.getPayloadSchema(dbusEvent);
 		Schema schema = writerSchema.getSchema();
 		DbusOpcode eventType = dbusEvent.getOpcode();
-		Set<String> primaryKeysSet = getPrimaryKeysSetFromSchema(schema);
+		Set<String> primaryKeysSet = AvroSchemaHelper.getPrimaryKeysSetFromSchema(schema);
 		String namespaceName = schema.getNamespace();
 		String entityName = schema.getName();
 		Map<String, Object> fieldMap = new HashMap<String, Object>();
+		Map <String, String> fieldToMysqlDataType = AvroSchemaHelper.fieldToDataTypeMap(schema);
+
 		for (Field field : schema.getFields())
 		{
-			String mysqlType = SchemaHelper.getMetaField(field, META_FIELD_TYPE_NAME);
-			fieldMap.put(
-			        field.name(),
-			        AvroToMysqlMapper.avroToMysqlType(genericRecord.get(field.name()),
-			                MysqlDataTypes.valueOf(mysqlType.toUpperCase())));
+			Object recordValue = genericRecord.get(field.name());
+			Object mysqlTypedObject;
+			if (AvroSchemaHelper.isRowChangeField(field))
+			{
+				mysqlTypedObject = getMysqlObjectForRowChangeField((HashMap<Object, Object>) recordValue, fieldToMysqlDataType);
```

Comment:
Shouldn't this be Map<String, Object> because the key in the changed old values map is always a string?